packages/ember-runtime/lib/mixins/-proxy.js

```diff
@@ -25,7 +25,7 @@ import symbol from 'ember-metal/symbol';
 const IS_PROXY = symbol('IS_PROXY');
 
 export function isProxy(value) {
-  return value && value[IS_PROXY];
+  return typeof value === 'object' && value && value[IS_PROXY];
```

Comment:
I think it may be better to do `typeof value === 'object' && value !== null` instead of coercing `value` to a boolean here.